WordPress/openverse-frontend
# Avoid error using only local translation files

## Fixes


Fixes #298 by @sarayourfriend

## Description


This addresses the problem of the project failing to run if not all translation files are present, avoiding doing `i18n:get-translations` previously.

It adds a step to the `dev` command to generate the locales list. The downside of this approach is that it modifies the `valid-locales.json` file and we will need to be alert in PR changes, but do we really need to track it? Looks like it is actually not used.

## Testing Instructions

1. Delete any translation file other than the `en.json`
2. Run the project in this branch: `npm run dev`
3. See no errors.

## Checklist


[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
